### PR TITLE
refactor: string colors to themes

### PR DESCRIPTION
**Pull request by tuliomir**
While investigating incorrect colors for the application on #326 , all the application existing UI colors were mapped.

This PR refactors all hardcoded colors spread throughout the code into the centralized theme constants. This will allow for easier future development of configurable themes and dark mode.

### Acceptance Criteria
- UI colors should use `themes.js` constants instead of harcoded strings

| :information_source: Changed colors |
| :--- |
| Some hardcoded colors were slightly off the mapped ones and were adjusted to the nearest available. These cases were highlighted in the comments below for easier review. |

### Security Checklist
- [X] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/components/HathorTextInput.js [hunk 1]
```diff
@@ -7,12 +7,17 @@
...
 
 const HathorTextInput = (props) => (
   <TextInput
     {...props}
     style={[{
-      width: 100, padding: 8, borderRadius: 4, borderColor: 'gainsboro', borderWidth: 1,
+      width: 100,
+      padding: 8,
+      borderRadius: 4,
+      borderColor: COLORS.borderColor,
```
Comment (tuliomir) [inline src/components/HathorTextInput.js:19]: Change from `#DCDCDC` to `#eeeeee`

File: src/components/QRCodeReader.js [hunk 2]
```diff
@@ -111,7 +112,7 @@ export default ({
         borderLeftWidth: borderWidth,
         borderRightWidth: borderWidth,
         backgroundColor: 'transparent',
-        borderColor: 'rgba(0, 0, 0, 0.7)',
+        borderColor: COLORS.textColorShadowDark,
```
Comment (tuliomir) [inline src/components/QRCodeReader.js:115]: Change from `0.7` opacity to `0.8` opacity

File: src/components/ReceiveMyAddress.js [hunk 3]
```diff
@@ -47,7 +49,7 @@ export default function ReceiveMyAddress() {
       padding: 16,
       borderBottomWidth: 1.5,
       borderTopWidth: 1.5,
-      borderColor: '#e5e5ea',
+      borderColor: COLORS.borderColor,
```
Comment (tuliomir) [inline src/components/ReceiveMyAddress.js:52]: Change from `#e5e5ea` to `#eeeeee`

File: src/components/ReceiveMyAddress.js [hunk 4]
```diff
@@ -90,10 +92,10 @@ const styles = StyleSheet.create({
     marginHorizontal: 16,
     marginTop: 32,
     borderWidth: 1.5,
-    borderColor: '#e5e5ea',
+    borderColor: COLORS.borderColor,
```
Comment (tuliomir) [inline src/components/ReceiveMyAddress.js:95]: Change from `#e5e5ea` to `#eeeeee`

File: src/components/TokenBox.js [hunk 5]
```diff
@@ -12,12 +12,13 @@ import {
...
 import { FontAwesomeIcon } from '@fortawesome/react-native-fontawesome';
 import { faSortDown } from '@fortawesome/free-solid-svg-icons';
+import { COLORS } from '../styles/themes';
 
 const TokenBox = (props) => (
   <TouchableWithoutFeedback onPress={props.onPress}>
     <View style={styles.wrapper}>
       <Text style={styles.label}>{props.label}</Text>
-      <FontAwesomeIcon icon={faSortDown} color='rgba(0, 0, 0, 0.7)' style={{ marginBottom: 5 }} />
+      <FontAwesomeIcon icon={faSortDown} color={COLORS.textColorShadowDark} style={{ marginBottom: 5 }} />
```
Comment (tuliomir) [inline src/components/TokenBox.js:21]: Change from `0.7` opacity to `0.8` opacity

File: src/components/WalletConnect/ModalButton.js [hunk 6]
```diff
@@ -11,20 +11,20 @@ const styles = StyleSheet.create({
     alignItems: 'center',
     marginLeft: 12,
     borderWidth: 2,
-    borderColor: '#cecece',
+    borderColor: COLORS.borderColor,
```
Comment (tuliomir) [inline src/components/WalletConnect/ModalButton.js:14]: Change from `#cecece` to `#eeeeee`

File: src/components/WalletConnect/ModalButton.js [hunk 7]
```diff
@@ -11,20 +11,20 @@ const styles = StyleSheet.create({
...
   },
   buttonHighlight: {
-    backgroundColor: PRIMARY_COLOR,
+    backgroundColor: COLORS.primary,
     borderWidth: null,
     borderColor: null,
   },
   buttonText: {
-    color: '#808080',
+    color: COLORS.midContrastDetail,
```
Comment (tuliomir) [inline src/components/WalletConnect/ModalButton.js:22]: Change from `#808080` to `#9e9e9e`

File: src/screens/WalletConnect/WalletConnectList.js [hunk 8]
```diff
@@ -49,7 +50,7 @@ const style = StyleSheet.create({
     flexDirection: 'row',
     alignItems: 'center',
     borderBottomWidth: 1,
-    borderBottomColor: '#cecece',
+    borderBottomColor: COLORS.lowContrastDetail,
```
Comment (tuliomir) [inline src/screens/WalletConnect/WalletConnectList.js]: Change from `#cecece` to `#eeeeee`

File: src/screens/WalletConnect/WalletConnectList.js [hunk 9]
```diff
@@ -67,13 +68,13 @@ const style = StyleSheet.create({
   },
   sessionName: {
     marginBottom: 8,
-    color: 'rgba(0, 0, 0, 0.87)',
+    color: COLORS.textColorShadowDark,
```
Comment (tuliomir) [inline src/screens/WalletConnect/WalletConnectList.js:71]: Changed from `0.87` opacity to `0.8` opacity

File: src/screens/WalletConnect/WalletConnectList.js [hunk 10]
```diff
@@ -67,13 +68,13 @@ const style = StyleSheet.create({
...
     fontSize: 12,
     fontWeight: 'bold',
   },
   sessionData: {
     marginBottom: 8,
-    color: 'rgba(0, 0, 0, 0.57)',
+    color: COLORS.textColorShadow,
```
Comment (tuliomir) [inline src/screens/WalletConnect/WalletConnectList.js:77]: Changed from `0.57` opacity to `0.5` opacity

File: src/components/NewHathorButton.js [hunk 11]
```diff
@@ -64,19 +65,19 @@ const style = StyleSheet.create({
   wrapper: {
     height: 48,
     borderRadius: 8,
-    backgroundColor: '#000',
+    backgroundColor: COLORS.textColor,
     alignSelf: 'stretch',
   },
   wrapperDisabled: {
-    backgroundColor: 'rgba(0, 0, 0, 0.05)',
+    backgroundColor: COLORS.textColorShadowLighter,
```
Comment (tuliomir) [inline src/components/NewHathorButton.js:72]: Changed from opacity `0.05` to `0.1`

File: src/components/WalletConnect/ApproveRejectModal.js [hunk 12]
```diff
@@ -15,10 +15,11 @@ import {
   StyleSheet,
 } from 'react-native';
 import ModalButton from './ModalButton';
+import { COLORS } from '../../styles/themes';
 
 const modalStyle = StyleSheet.create({
   signMessageText: {
-    backgroundColor: 'rgba(0, 0, 0, 0.12)',
+    backgroundColor: COLORS.textColorShadowLighter,
```
Comment (tuliomir) [inline src/components/WalletConnect/ApproveRejectModal.js:22]: Changed opacity from `0.12` to `0.1`

File: src/components/WalletConnect/SignMessageModal.js [hunk 13]
```diff
@@ -10,10 +10,11 @@ import { useDispatch } from 'react-redux';
 import { t } from 'ttag';
 import { StyleSheet, Text } from 'react-native';
 import ApproveRejectModal from './ApproveRejectModal';
+import { COLORS } from '../../styles/themes';
 
 const modalStyle = StyleSheet.create({
   signMessageText: {
-    backgroundColor: 'rgba(0, 0, 0, 0.12)',
+    backgroundColor: COLORS.textColorShadowLighter,
```
Comment (tuliomir) [inline src/components/WalletConnect/SignMessageModal.js:17]: Changed opacity from `0.12` to `0.1`

File: src/screens/WalletConnect/WalletConnectList.js [hunk 14]
```diff
@@ -49,7 +50,7 @@ const style = StyleSheet.create({
     flexDirection: 'row',
     alignItems: 'center',
     borderBottomWidth: 1,
-    borderBottomColor: '#cecece',
+    borderBottomColor: COLORS.borderColor,
```
Comment (tuliomir) [inline src/screens/WalletConnect/WalletConnectList.js:53]: Changed color from `#cecece` to `#eeeeee`

**Review by andreabadesso — APPROVED** [review]

File: src/components/HathorHeader.js [hunk 15]
```diff
@@ -61,7 +59,7 @@ const HathorHeader = (props) => {
 
   let extraStyle = {};
   if (props.withBorder) {
-    extraStyle = { borderBottomWidth: STYLE.borderWidth };
+    extraStyle = { borderBottomWidth: 1 };
```
Comment (pedroferreira1) [inline src/components/HathorHeader.js:62]: Why is it hardcoded as 1 and not using the style?

File: src/components/HathorTextInput.js [hunk 16]
```diff
@@ -7,12 +7,17 @@
 
 import React from 'react';
 import { TextInput } from 'react-native';
+import { COLORS, STYLE } from '../styles/themes';
```
Comment (pedroferreira1) [inline src/components/HathorTextInput.js:10]: Remove `STYLE` import?

File: src/components/HathorTextInput.js [hunk 17]
```diff
@@ -7,12 +7,17 @@
...
 const HathorTextInput = (props) => (
   <TextInput
     {...props}
     style={[{
-      width: 100, padding: 8, borderRadius: 4, borderColor: 'gainsboro', borderWidth: 1,
+      width: 100,
+      padding: 8,
+      borderRadius: 4,
+      borderColor: COLORS.borderColor,
+      borderWidth: 1,
```
Comment (pedroferreira1) [inline src/components/HathorTextInput.js:20]: Why hardcoded?

File: src/components/NewHathorButton.js (hunk 11 above)
Comment (pedroferreira1) [reply src/components/NewHathorButton.js:72]: Why?

File: src/components/QRCodeReader.js (hunk 2 above)
Comment (pedroferreira1) [reply src/components/QRCodeReader.js:115]: Why?

File: src/components/ReceiveMyAddress.js [hunk 18]
```diff
@@ -16,14 +16,16 @@ import { useDispatch, useSelector } from 'react-redux';
...
 
 export default function ReceiveMyAddress() {
   const dispatch = useDispatch();
   const wallet = useSelector((state) => state.wallet);
   const lastSharedAddress = useSelector((state) => state.lastSharedAddress);
 
   const getNextAddress = async () => {
-    const { address, index } = await wallet.getNextAddress();
+    // Fetch the address but do not mark it as used yet.
+    const { address, index } = await wallet.getCurrentAddress();
```
Comment (pedroferreira1) [inline src/components/ReceiveMyAddress.js:28]: What's this change? Isn't this PR just for color changes? 

And I think this is not correct, we should get the next address, not current.

File: src/components/ReceiveMyAddress.js (hunk 3 above)
Comment (pedroferreira1) [reply src/components/ReceiveMyAddress.js:52]: Why?

File: src/components/SimpleInput.js [hunk 19]
```diff
@@ -70,7 +70,7 @@ const SimpleInput = (props) => {
 
 const styles = StyleSheet.create({
   inputContainer: {
-    borderBottomWidth: STYLE.borderWidth,
+    borderBottomWidth: 1,
```
Comment (pedroferreira1) [inline src/components/SimpleInput.js:73]: Why hardcoded?

File: src/components/WalletConnect/ModalButton.js (hunk 6 above)
Comment (pedroferreira1) [reply src/components/WalletConnect/ModalButton.js:14]: Why?

File: src/components/WalletConnect/ModalButton.js (hunk 7 above)
Comment (pedroferreira1) [reply src/components/WalletConnect/ModalButton.js:22]: Why don't we keep the same color?

File: src/screens/WalletConnect/WalletConnectList.js (hunk 14 above)
Comment (pedroferreira1) [reply src/screens/WalletConnect/WalletConnectList.js:53]: Why?

File: src/screens/WalletConnect/WalletConnectList.js (hunk 9 above)
Comment (pedroferreira1) [reply src/screens/WalletConnect/WalletConnectList.js:71]: Do we have a constant with `0.9`?

File: src/screens/WalletConnect/WalletConnectList.js (hunk 10 above)
Comment (pedroferreira1) [reply src/screens/WalletConnect/WalletConnectList.js:77]: Do we have a constant with `0.6`?

File: src/components/ReceiveMyAddress.js (hunk 18 above)
Comment (tuliomir) [reply src/components/ReceiveMyAddress.js]: This escaped from the latest rebase with `dev`, thanks for pointing out!
Fixed on [e18e360](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/327/commits/e18e3609ae83e87c8081a4c265db2e93276965be).

File: src/components/HathorTextInput.js (hunk 16 above)
Comment (tuliomir) [reply src/components/HathorTextInput.js]: Done on [2014bd3](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/327/commits/2014bd3e1494a2c753663eab2afc5baf1937f7f5)

File: src/components/HathorHeader.js (hunk 15 above)
Comment (tuliomir) [reply src/components/HathorHeader.js:62]: Actually the style was removed, as on 326 it was resolved that we would no longer have different border widths for Android and iOS.

Adding the many variations of each component's borders would not improve maintainability of the code, so I found it best to just let these border widths be hardcoded again.

File: src/components/HathorTextInput.js (hunk 17 above)
Comment (tuliomir) [reply src/components/HathorTextInput.js:20]: See comment above about border widths

File: src/components/SimpleInput.js (hunk 19 above)
Comment (tuliomir) [reply src/components/SimpleInput.js:73]: See comment above about border widths

File: src/components/QRCodeReader.js (hunk 2 above)
Comment (tuliomir) [reply src/components/QRCodeReader.js]: Reverted to 0.8 opacity on [a6b16fd](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/327/commits/a6b16fdf489a4652038ac614da23fe5485e62020)

File: src/components/ReceiveMyAddress.js (hunk 3 above)
Comment (tuliomir) [reply src/components/ReceiveMyAddress.js:51]: As discussed, this is close enough to the current border color to compose the app theme.

File: src/components/WalletConnect/ModalButton.js (hunk 6 above)
Comment (tuliomir) [reply src/components/WalletConnect/ModalButton.js:14]: As discussed, this is close enough to the current border color to compose the app theme.

File: src/screens/WalletConnect/WalletConnectList.js (hunk 14 above)
Comment (tuliomir) [reply src/screens/WalletConnect/WalletConnectList.js:53]: As discussed, this is close enough to the current border color to compose the app theme.

File: src/components/TokenBox.js (hunk 5 above)
Comment (tuliomir) [reply src/components/TokenBox.js]: Reverted to `0.7` opacity on [a6b16fd](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/327/commits/a6b16fdf489a4652038ac614da23fe5485e62020)

File: src/components/WalletConnect/ModalButton.js (hunk 7 above)
Comment (tuliomir) [reply src/components/WalletConnect/ModalButton.js]: Reverted to original color on [a6b16fd](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/327/commits/a6b16fdf489a4652038ac614da23fe5485e62020)

File: src/screens/WalletConnect/WalletConnectList.js (hunk 9 above)
Comment (tuliomir) [reply src/screens/WalletConnect/WalletConnectList.js]: Reverted to `0.9` opacity on [a6b16fd](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/327/commits/a6b16fdf489a4652038ac614da23fe5485e62020)

File: src/screens/WalletConnect/WalletConnectList.js (hunk 10 above)
Comment (tuliomir) [reply src/screens/WalletConnect/WalletConnectList.js]: Changed to `0.6` opacity on [a6b16fd](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/327/commits/a6b16fdf489a4652038ac614da23fe5485e62020)

File: src/components/NewHathorButton.js (hunk 11 above)
Comment (tuliomir) [reply src/components/NewHathorButton.js]: Reverted to `0.05` opacity on [a6b16fd](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/327/commits/a6b16fdf489a4652038ac614da23fe5485e62020)

File: src/components/HathorTextInput.js (hunk 1 above)
Comment (tuliomir) [reply src/components/HathorTextInput.js]: Reverted to original color on [2e29c07](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/327/commits/2e29c070354d1fae1ee605c2884a2c6ff80cb0b2)

File: src/components/WalletConnect/ModalButton.js (hunk 6 above)
Comment (tuliomir) [reply src/components/WalletConnect/ModalButton.js]: Reverted to original color on [2e29c07](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/327/commits/2e29c070354d1fae1ee605c2884a2c6ff80cb0b2)

File: src/screens/WalletConnect/WalletConnectList.js (hunk 14 above)
Comment (tuliomir) [reply src/screens/WalletConnect/WalletConnectList.js]: Reverted to original color on [2e29c07](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/327/commits/2e29c070354d1fae1ee605c2884a2c6ff80cb0b2)

**Review by pedroferreira1 — APPROVED** [review]

**Pull request merged**